### ISSUE TITLE
Rainmeter WebNowPlaying - Cover Art Blinks Randomly

### ISSUE THREAD

**Issue by remorsecodex** [ytmdesktop/ytmdesktop#439]
- [x] I understand that **YTMDesktop have NO affiliation with Google or YouTube**.
- [x] I verified that there is no open issue for the same subject.

**Describe the bug**

I just set up YTMDesktop integration via WebNowPlaying with RainMeter for my desktop where I use Monstercat Visualizer to show album art, spectrometer, and track info. It all works fine except that the cover art blinks (momentarily shows the default Monstercat cover art) at random. 

The shortest time between blinks that I've timed is about 3s, and the longest is about 45s.

**To Reproduce**
1. Play music in YTMDesktop with WebNowPlaying reporting to another app/integration that displays cover art.
2. Observe blinking at random.

**Expected behavior**
Cover art to show without blinking like this - steady, only changing when albums change while playing music.

**Environment (please complete the following information):**
 * YTMDesktop version: 1.13.0
 * OS: Windows 10 
 * OS version: Enterprise N v 10.0.19041
 * Arch: x64
* Installation way: exe

**Comment by remorsecodex** [comment]
Update: Checked the rainmeter logs and found this:

ERRO (20:39:12.018) monstercat-visualizer\visualizer.ini - [MeasureWebParser]: (Fetch error) The operation timed out  (ErrorCode=12002)
ERRO (20:39:34.921) : Unable to download album art to: C:\Users\Username\Documents\Rainmeter\Skins\monstercat-visualizer\@Resources\images\coverWeb.png

**Comment by tjhrulz** [comment]
Copying a discussion I had about this issue here since it was brought to my attention:

This is the first I have heard of this application, however looking over its source code this application seems to be emulating the extension and making a connection to WebNowPlaying's Rainmeter plugin. I will have to look into the exact reason it is causing this issue, if I had to guess from a quick glance they do not have how the extension communicates 100% right and are sending this data constantly and not just when a change has been detected.

To elaborate further the caching for new information is done of the client side of the websocket this is done to avoid sending an overabundance of information over the websocket, while I can't be certain as I do not have a devenv set up where I can compile this program myself but looking over the code it would seem that it is constantly sending that data and thus constantly updating the cover art. Likely it is sending so many download requests that the Rainmeter plugin is getting blocked from downloading it again and it times out.


I have added an issue on my end to look into adding a check to the plugin to prevent redownloading album art that is the one most recently downloaded however I recommend you also add caching to ytmdesktop so that we can prevent any other idiosyncrasies between the two programs (As well as save the precious CPU cycles and networking overhead :P )

https://github.com/tjhrulz/WebNowPlaying/issues/50

**Comment by tjhrulz** [comment]
Since the cause of this has not been fixed yet in this app I have fixed the cause on my end of the WebNowPlaying plugin, so as long as people have version 0.5.0.0 they will not encounter this bug.

However I still strongly encourage you to cache info in the app and only send the info as it changes as that was what caused this in the first place.

**Comment by Alipoodle** [comment]
This is a dupe of #283
TL;DR Slow down the sending of information.
